src/main/scala/play/api/cache/redis/configuration/RedisConfigLoader.scala

```diff
@@ -15,11 +15,23 @@ private[ configuration ] object RedisConfigLoader {
 
   implicit class ConfigOption( val config: Config ) extends AnyVal {
     def getOption[ T ]( path: String, f: Config => String => T ): Option[ T ] =
-      if ( config hasPath path ) Some( f( config )( path ) ) else None
+      config match {
+        case c if c.hasPath(path) => Some(f( c )( path ))
+        case _ => None
+      }
   }
 
   implicit class ConfigPath( val path: String ) extends AnyVal {
-    def /( suffix: String ): String = if ( path == "" ) suffix else s"$path.$suffix"
+
+    private def eagerDecision(): Function1[String, String] =
+      path.isEmpty match {
+        case true => (suffix: String) => suffix
+        case _ => (suffix: String) => path + "." + suffix
+      }
+
+    private def delegate: Function1[String, String] = eagerDecision()
+
+    def /( suffix: String ): String = delegate(suffix)
```

Comment:
This change significantly reduces the code readability a complicates its maintenance and I don't see any added value without 10 times more SLOC